File: JavaScript/TestXS.js
"use strict";

const XorShift=require("./XorShift");

(function(){
	//XorShift乱数ジェネレータの初期化
	// 論文デフォルトシード
	const r_def=new XorShift.defaultSeed();
	// 固定値シード
	const r_const=new XorShift(100);
	// 時間シード
	const r=new XorShift();

	//デフォルトシード値の取得
	console.log(">> defaults");
	console.log(XorShift.defaults);

	//適用したシード値の取得
	console.log(">> seeds in r");
	console.log(r.seeds);

	//乱数の生データを取得
	console.log(">> rand 0 to UInt32Max");
	for(let i=0;i<5;i++){
		console.log(r_def.rand());
	}

	//0-100の乱数(100含む)を整数で取得
	console.log(">> randInt 0 to 100");
	for(let i=0;i<5;i++){
		console.log(r_const.randInt(0,100));
	}

	//0-1の乱数を浮遊小数点で取得
	console.log(">> randFloat 0 to 1");
	for(let i=0;i<5;i++){
		console.log(r.randFloat());
	}

	//配列のシャッフル
	//値渡しとなるので元の配列は破壊されない
	console.log(">> shuffle Array");
	var a=[...Array(20).keys()];
	console.log(r.shuffle(a));
	console.log(a);

	//今の乱数を回した回数
	console.log(">> randCount in r");
	console.log(r.randCount);
})();

File: Clojure/TestXS.clj
(require 'XorShift)

(let [
	r_def (XorShift/make-XorShift-defaultSeed)
	r_const (XorShift/make-XorShift 100)
	r (XorShift/make-XorShift)
	a (atom nil)
]
	;デフォルトのシード値を取得
	(println ">> defaults")
	(println XorShift/defaults)

	;今の乱数に与えたシード値初期値
	(println ">> seeds in r")
	(println (:seeds r))

	;乱数の生データを取得
	(println ">> rand 0 to UInt32Max")
	(loop[i 0](when(< i 5)
		(println (XorShift/-rand r_def))
	(recur(inc i))))

	;0-100の乱数(100含む)を整数で取得
	(println ">> randInt 0 to 100")
	(loop[i 0](when(< i 5)
		(println (XorShift/randInt r_const 0 100))
	(recur(inc i))))

	;0-1の乱数を浮遊小数点で取得
	(println ">> randFloat 0 to 1")
	(loop[i 0](when(< i 5)
		(println (XorShift/randFloat r))
	(recur(inc i))))

	;listのシャッフル
	;値渡しとなるので元の配列は破壊されない
	(println ">> shuffle list")
	(reset! a (for[i (range 20)] i))
	(println (XorShift/-shuffle r @a))
	(println @a)

	;今の乱数を回した回数
	(println ">> randCount in r")
	(println @(:randCount r))
)